src/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs

```diff
@@ -669,13 +669,70 @@ private object WritePrimitive(TypeMapping mapping, Func<string> readFunc)
                 string methodName = $"To{mapping.TypeDesc.FormatterName}";
                 if (!mapping.TypeDesc.HasCustomFormatter)
                 {
-                    MethodInfo method = typeof(XmlConvert).GetMethod(methodName, BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic, new Type[] { typeof(string) });
```

Comment:
Good question. I just checked all `XmlConvert.ToXXXX` methods. Two methods are internal, `ToBinHexString` and `ToInteger`. But neither `BinHexString` nor `Integer` is valid FormatterName (for interger the FormatterName is `Int32`). So we shouldn't have invoked these two methods from here. The `BindingFlags.NonPublic` might be misused.